src/System.Runtime/tests/System/StringTests.netcoreapp.cs

```diff
@@ -39,5 +40,82 @@ public static void StartsWith(string s, char value, bool expected)
         {
             Assert.Equal(expected, s.StartsWith(value));
         }
+
+        public static IEnumerable<object[]> Join_Char_StringArray_TestData()
+        {
+            yield return new object[] { '|', new string[0], 0, 0, "" };
+            yield return new object[] { '|', new string[] { "a" }, 0, 1, "a" };
+            yield return new object[] { '|', new string[] { "a", "b", "c" }, 0, 3, "a|b|c" };
+            yield return new object[] { '|', new string[] { "a", "b", "c" }, 0, 2, "a|b" };
+            yield return new object[] { '|', new string[] { "a", "b", "c" }, 1, 1, "b" };
+            yield return new object[] { '|', new string[] { "a", "b", "c" }, 1, 2, "b|c" };
+            yield return new object[] { '|', new string[] { "a", "b", "c" }, 0, 0, "" };
+            yield return new object[] { '|', new string[] { "", "", "" }, 0, 3, "||" };
+            yield return new object[] { '|', new string[] { null, null, null }, 0, 3, "||" };
+        }
+
+        [Theory]
+        [MemberData(nameof(Join_Char_StringArray_TestData))]
+        public static void Join_Char_StringArray(char separator, string[] values, int startIndex, int count, string expected)
+        {
+            if (startIndex == 0 && count == values.Length)
+            {
+                Assert.Equal(expected, string.Join(separator, values));
+                Assert.Equal(expected, string.Join(separator, (IEnumerable<string>)values));
+                Assert.Equal(expected, string.Join(separator, (object[])values));
+                Assert.Equal(expected, string.Join(separator, (IEnumerable<object>)values));
+            }
+
+            Assert.Equal(expected, string.Join(separator, values, startIndex, count));
```

Comment:
To help verify that the behavior is the same with the string overloads, how about adding:
```C#
Assert.Equal(expected, string.Join(separator.ToString(), values, startindex, count));
```